Sources/QuickObjectiveC/QuickSpec.m

```diff
@@ -109,22 +111,24 @@ + (void)buildExamplesIfNeeded {
  @return The selector of the newly defined instance method.
  */
 + (SEL)addInstanceMethodForExample:(Example *)example classSelectorNames:(NSMutableSet<NSString*> *)selectorNames {
-    IMP implementation = imp_implementationWithBlock(^(QuickSpec *self){
+    IMP implementation = imp_implementationWithBlock(^(QuickSpec *self, void (^completionHandler)(void)){
         self.example = example;
         currentSpec = self;
-        [example run];
+        [example runWithCompletionHandler:completionHandler];
     });
 
-    const char *types = [[NSString stringWithFormat:@"%s%s%s", @encode(void), @encode(id), @encode(SEL)] UTF8String];
+    const char *types = [[NSString stringWithFormat:@"%s%s%s@?<v@?>", @encode(void), @encode(id), @encode(SEL)] UTF8String];
 
     NSString *originalName = [QCKObjCStringUtils c99ExtendedIdentifierFrom:example.name];
     NSString *selectorName = originalName;
     NSUInteger i = 2;
-    
-    while ([selectorNames containsObject:selectorName]) {
+
+    while ([selectorNames containsObject:[selectorName stringByAppendingFormat:@"%s", @encode(SEL)]]) {
```

Comment:
Now that all the test names take in a single argument, we need to also append the ":" when we check for existing selectors.